### PR TITLE
Sentry SDK setup

### PR DESCRIPTION
**Pull request by jos-b**
Implement the Sentry SDK to the bot.

This PR will also setup basic tags to link errors together.

### CODE REVIEW

**Comment by jos-b** [comment]
image

**Comment by jos-b** [comment]
This PR has also had to remove the API logging handler (on our plan anyway) since it was blocking sentry from making it's API calls.

File: bot/__main__.py [hunk 1]
```diff
@@ -1,10 +1,16 @@
...
 from discord.ext.commands import when_mentioned_or
+from sentry_sdk.integrations.aiohttp import AioHttpIntegration
 
 from bot import patches
 from bot.bot import Bot
 from bot.constants import Bot as BotConfig, DEBUG_MODE
 
+sentry_sdk.init(
+    dsn=BotConfig.sentry_dsn,
+    integrations=[AioHttpIntegration()]
```
Comment (MarkKoz) [inline bot/__main__.py]: Is this going to complain about not being created inside an async function?

Comment (jos-b) [reply bot/__main__.py]: It won't but I have just realised this is intended for aiohttp servers, not clients.

https://docs.sentry.io/platforms/python/aiohttp/

File: bot/cogs/error_handler.py [hunk 2]
```diff
@@ -147,10 +148,18 @@ def __init__(self, bot: Bot):
             f"Sorry, an unexpected error occurred. Please let us know!\n\n"
             f"```{e.__class__.__name__}: {e}```"
         )
-        log.error(
-            f"Error executing command invoked by {ctx.message.author}: {ctx.message.content}"
-        )
-        raise e
```
Comment (MarkKoz) [inline bot/cogs/error_handler.py]: Is this still going to be logged to stdout? (by the way, in #757 I changed this to use `exec_info` instead of raising)

Comment (jos-b) [reply bot/cogs/error_handler.py]: log.error should go to stdout shouldn't it?

Comment (MarkKoz) [reply bot/cogs/error_handler.py]: Well yes but the point is that you've completely removed these lines.

Comment (jos-b) [reply bot/cogs/error_handler.py]: Will use exc_info though, I like that

Comment (jos-b) [reply bot/cogs/error_handler.py]: Oh wait, I didn't understand this right, I removed the log.error, adding back.

Comment (MarkKoz) [reply bot/cogs/error_handler.py]: Since you're still doing `capture_exception`, will Sentry not end up with duplicate entries for this exception?

Comment (jos-b) [reply bot/cogs/error_handler.py]: Yeah, I've removed the capture_exception and added log.error:

image

image

Comment (MarkKoz) [reply bot/cogs/error_handler.py]: It still gets those extra tags set?

Comment (jos-b) [reply bot/cogs/error_handler.py]: Yeah, because I run the log command from inside a scope I've created

**Comment by MarkKoz** [comment]
Is this ready to be merged @jos-b? Is the environment variable set in prod?

**Comment by jos-b** [comment]
@SebastiaanZ It's something which we can adapt as we find a workflow which suits us well. Right now just storing errors somewhere is useful and we'll find out if we need anything else as time goes on.

@MarkKoz the variable is in prod 👍

**Pull request merged**